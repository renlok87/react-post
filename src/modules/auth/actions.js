export const RECIEVE_AUTH = 'auth/RECIEVE_AUTH';
export const LOGIN = 'auth/LOGIN';
export const LOGOUT = 'auth/LOGOUT';
