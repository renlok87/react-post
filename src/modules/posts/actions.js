import moment from 'moment';
import { v4 as uuid } from 'uuid';

const scope = '@@posts';

export const CREATE_POST = `${scope}/CREATE_POST`;

export const createPost = payload => ({
  type: CREATE_POST,
  payload: {
    id: uuid(),
    createdAt: moment(),
    userId: null,
    text: '',
    replyToId: null,
    ...payload,
  },
});
