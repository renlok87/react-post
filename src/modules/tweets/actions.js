export const CREATE_TWEET = 'tweets/CREATE_TWEET';
