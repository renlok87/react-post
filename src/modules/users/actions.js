export const CREATE_USER = 'users/CREATE_USER';
