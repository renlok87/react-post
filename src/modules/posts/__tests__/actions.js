import * as actions from '../actions';

jest.mock('uuid', () => ({
  v4: () => '<id>',
}));

jest.mock('moment', () => () => '<datetime>');

describe('posts actions', () => {
  it('exports constants', () => {
    expect(actions.CREATE_POST).toBeDefined;
  });

  it('creates post', () => {
    const post = actions.createPost({
      text: 'Test post',
    });

    expect(post).toMatchSnapshot();
  });
});
