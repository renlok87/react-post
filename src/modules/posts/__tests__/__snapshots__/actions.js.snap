// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweets actions creates post 1`] = `
Object {
  "payload": Object {
    "createdAt": "<datetime>",
    "id": "<id>",
    "replyToId": null,
    "text": "Test post",
    "userId": null,
  },
  "type": "@@tweets/CREATE_POST",
}
`;
